### PR TITLE
spidershim: Partially implement isolate terminate execution

### PR DESCRIPTION
**Pull request by brendandahl**
This will need more work when object/function template is implemented.

I filed bugs #95 and #96 and wrote more in https://github.com/mozilla/spidernode/issues/85#issuecomment-219587766

### CODE REVIEW

File: deps/spidershim/src/v8isolate.h [hunk 1]
```diff
@@ -51,6 +51,8 @@ struct Isolate::Impl {
   mozilla::Maybe<internal::RootStore> eternals;
   std::vector<MessageCallback> messageListeners;
   void* embeddedData[internal::kNumIsolateDataSlots];
+  mozilla::Atomic<bool> serviceInterrupt;
+  mozilla::Atomic<bool> terminatingExecution;
```
Comment (ehsan) [inline deps/spidershim/src/v8isolate.h]: Out of curiosity, why did you use atomics here?  This is all called on the same thread, right?

File: deps/spidershim/test/isolate.cc [hunk 2]
```diff
@@ -96,3 +97,101 @@ TEST(SpiderShim, IsolateEmbedderData) {
...
+  args.rval().setUndefined();
+  return true;
+}
+
+bool Fail(JSContext *cx, unsigned argc, JS::Value *vp) {
+  EXPECT_TRUE(false);
+  return true;
+}
+
+Local<Value> CompileRun(const char* script) {
```
Comment (ehsan) [inline deps/spidershim/test/isolate.cc:115]: Can you please use `engine.CompileRun()` instead and take this out?

**Comment by ehsan** [comment]
Looks good!  I submitted a couple of minor nits.

File: deps/spidershim/src/v8isolate.h (hunk 1 above)
Comment (brendandahl) [inline deps/spidershim/src/v8isolate.h]: I was following some spider monkey code that does this, thinking that we'd need to support workers/threads in the future. Now looking into it, it seems electron does not support workers.

**Pull request merged**